### PR TITLE
Render Vertical Participants and Lanes

### LINKED ISSUE #57: Correctly render pool / lane labels in vertical collaboration
__Describe the issue__

Vertical pools is a feature request we see once in a while. It is also part of [the MIWG test suite, case `A.4.1`](https://github.com/bpmn-io/bpmn-miwg-test-suite/blob/master/Reference/A.4.1.bpmn) specifically.

We do not account for pool direction when rendering it, resulting in such diagrams: 

a 4 1-import

Properly supporting vertical pools does require modeling support (https://github.com/bpmn-io/bpmn-js/issues/507), too.

__Additional Details__

A pool is marked as vertical via the `isHorizontal="false"` property in the element's DI:

```xml
         <bpmndi:BPMNShape bpmnElement="sid-66751F1E-EEB9-4BA7-9FDA-7965A1CA9CD1" id="sid-66751F1E-EEB9-4BA7-9FDA-7965A1CA9CD1_gui" isHorizontal="false">
            <omgdc:Bounds height="678.5" width="190.00000000000003" x="214.2857142857143" y="72.5"/>
            <bpmndi:BPMNLabel labelStyle="sid-ae9a9300-bd5c-4531-84ab-5f0791d9f49f">
               <omgdc:Bounds height="12.0" width="36.857147216796875" x="290.8571406773159" y="78.5"/>
            </bpmndi:BPMNLabel>
         </bpmndi:BPMNShape>
```

---

See [reference image](https://github.com/bpmn-io/bpmn-miwg-test-suite/blob/master/Reference/A.4.1.png) for the picture above.

### PR DESCRIPTION
**Pull request by sombrek**
Hi there,

I propose an implementation for rendering vertical pools and lanes.
Fixes #57 

The labels of vertical pools and lanes as well as the pool label separator are now correctly rendered on top.
Labels of vertical black-box pools are rotated.

The spec implies that horizontal is the default and I adhere to that, i.e. a missing isHorizontal attribute defaults to true.

Could you please provide some information how I can test this?
I thought about adding a vertical version of https://github.com/bpmn-io/bpmn-js/blob/7aee439f47cf0daebcb6131e84f0f9a9d2a51544/test/fixtures/bpmn/draw/pools.bpmn
...but I'm not quite sure if that'll do it.

Kind regards

<details>
<summary>Showcase</summary>

showcase

</details>

### CODE REVIEW

**Comment by philippfromme** [comment]
Thank you for your contribution. Why is this a draft? Are you still working on it?

**Comment by sombrek** [comment]
Thank you for your feedback.

The suggested test has been added nd it passes.
I also adjusted the IsHorizontalFix. However, I believe this affects only modeling, but not rendering.

> `isHorizontal` has no default value

The spec is funny. Pools and lanes shall be horizontal, if either `isVertical` is `false` or missing (BPMN 2.0.2, page 402). However no other mention of `isVertical` appears in the spec, so I thought they changed it but forgot to update that part of the spec.

Anyway, is this the point where I should mark the pull request as ready?

File: lib/features/modeling/behavior/IsHorizontalFix.js [hunk 1]
```diff
@@ -35,10 +35,15 @@ export default function IsHorizontalFix(eventBus) {
...
+    if (isAny(bo, elementTypesToUpdate)) {
+      var isHorizontal = di.get('isHorizontal');
+
+      if (isHorizontal === undefined) {
+        isHorizontal = true;
+      }
 
       // set attribute directly to avoid modeling#updateProperty side effects
-      di.set('isHorizontal', true);
+      di.set('isHorizontal', isHorizontal);
```
Comment (nikku) [inline lib/features/modeling/behavior/IsHorizontalFix.js:46]: Let's add a test case to verify this is being set to the proper thing, on move.

**Comment by philippfromme** [comment]
Label editing might have to be adjusted. This could be a follow-up.

brave_igjo7RaiZq

**Review by nikku — CHANGES_REQUESTED** [review]
@sombrek We'd need to adjust label editing [here](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/features/label-editing/LabelEditingProvider.js#L279) + add a test case.

**Comment by sombrek** [comment]
@nikku and @philippfromme Thanks for your support.

I'd like to ask you to reconsider your latest remarks as I feel they extend the scope of this pull request too much.

In this pull request I specifically wanted to target vertical rendering, i.e. issue #57 only.
The issue tracker separately lists issue #507 for vertical modeling. I consider labeling and moving stuff around modeling, so I didn't have a look at that yet. I probably diluted my intent by implementing the changes to `IsHorizontalFix`. Sorry for that.

As suggested earlier, I added test cases that ensure the attribute remains intact after moving.
Are you willing to accept that as a complete increment or do you prefer another approach?

**Comment by nikku** [comment]
@sombrek Thanks for getting back to us. We believe that supporting a feature end-to-end, especially as the remaining steps are low effort, is valuable.

You're right, you planned to contribute _rendering_. We'll only incorporate rendering once _basic modeling_ works, too.

As I understand you you don't have capacity to work on the modeling part? If that is the case then we're happy to help out. Note that this may take a little bit longer then to get this merged.

**Comment by sombrek** [comment]
@nikku Labeling now works as expected. Thanks for pointing me to the code.

> Fun fact: Label editing was our first modeling feature, before we could do anything else.

Well, you've got tho give the baby a name.

**Comment by dymkkk** [comment]
I implemented a demo, but there were some issues with the lane when adjusting the size
[demo](http://test.cn-wbst.cn/dym/bpmn/)

**Comment by sombrek** [comment]
@dymkkk Thanks for trying.

This issue was only about rendering (drawing) vertical pools lanes.

I'm currently working on resize issues. Those are documented in issue #2062.
There are more issues that I plan to work on after that.
A summary of what's not working yet can be found in issue #507.

**Comment by dymkkk** [comment]
> @dymkkk Thanks for trying.
> 
> This issue was only about rendering (drawing) vertical pools lanes.
> 
> I'm currently working on resize issues. Those are documented in issue #2062. There are more issues that I plan to work on after that. A summary of what's not working yet can be found in issue #507.

This demo covers more than just that. You probably didn't try to create swim lanes. Swimlanes can already be split and created vertically, and can be resized, but there are some problems with range limitations. I also noticed your plan. If you are continuing to do it, then I still look forward to your results. Thanks

**Pull request merged**